plugins/reversed_selection_listener.py

```diff
@@ -0,0 +1,38 @@
+import sublime
+import sublime_plugin
+
+# Related reading:
+#   https://discord.com/channels/280102180189634562/280102180189634562/958470369767997500
+#
+# This plugin provides a custom key binding context named reversed_selection
+# which can be used to create key bindings that trigger only when the selection
+# is "normal" (cursor on the right) or reversed (cursor on the left).
+#
+# An example of this in use is the following key binding, which will echo an
+# empty dictionary to the console if the key is pressed and all selections are
+# reversed.
+#
+# { "keys": ["ctrl+alt+c"], "command": "echo",
+#   "context": [
+#     { "key": "reversed_selection", "operator": "equal", "operand": true, "match_all": true, },
+#    ]
+# },
+
+class ReverseSelectionListener(sublime_plugin.EventListener):
+    """
+    This event listener listens for a key context named reversed_selection and
+    will return True or False depending on whether the selection is selected in
+    a reverse manner or not. match_all is also supported, to allow the context
+    to work across multiple cursors.
+    """
+    def on_query_context(self, view, key, operator, operand, match_all):
+        if key != "reversed_selection":
+            return None
+
+        lhs = operand
+        if match_all:
+            rhs = all([r.a > r.b for r in view.sel()])
+        else:
+            rhs = view.sel()[0].a > view.sel()[0].b
+
+        return True if operator == sublime.OP_EQUAL and rhs == lhs else False
```

Comment:
what if OP_NOT_EQUAL and rhs != lhs - that should also be true, right?

Reply:
It turns out that this is more complicated than that, because if you're comparing for False (even when the operator is EQUALS), the state of the values in the array needs to be reversed.

Otherwise, the `any()` or `all()` tests for `True` but only finds `False`, and thinks that it should not trigger when in fact it should.

This one seems to work for all possible cases of the two operators and `match_all`.

I resolved the underlying issue by flipping the state of the operand when the operator is `!=` so that it always works out the same as an `==` operator will.